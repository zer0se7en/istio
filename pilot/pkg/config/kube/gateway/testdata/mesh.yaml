apiVersion: networking.x-k8s.io/v1alpha1
kind: GatewayClass
metadata:
  name: istio
spec:
  controller: istio.io/gateway-controller
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: Gateway
metadata:
  name: gateway
  namespace: istio-system
spec:
  gatewayClassName: istio
  listeners:
  - hostname: "*.example.com"
    port: 80
    protocol: HTTP
    routes:
      namespaces:
        from: All
      kind: HTTPRoute
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: HTTPRoute
metadata:
  name: echo
  namespace: default
spec:
  gateways:
    allow: FromList
    gatewayRefs:
      - name: mesh
        namespace: istio-system
  hostnames: ["echo.default.svc.cluster.local"]
  rules:
  - forwardTo:
    - serviceName: echo
      port: 80
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: HTTPRoute
metadata:
  name: dual # applies to mesh and explicit gateway
  namespace: default
spec:
  gateways:
    allow: FromList
    gatewayRefs:
      - name: mesh
        namespace: istio-system
      - name: gateway
        namespace: istio-system
  hostnames: ["foo.example.com"]
  rules:
  - forwardTo:
    - serviceName: example
      port: 80
