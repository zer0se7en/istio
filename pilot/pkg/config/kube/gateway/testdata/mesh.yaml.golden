apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway
  namespace: istio-system
spec:
  servers:
  - hosts:
    - '*.example.com'
    port:
      name: 0-gateway-gateway-istio-system
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  creationTimestamp: null
  name: echo-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - mesh
  hosts:
  - echo.default.svc.cluster.local
  http:
  - route:
    - destination:
        host: echo.default.svc.domain.suffix
        port:
          number: 80
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  creationTimestamp: null
  name: dual-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway
  - mesh
  hosts:
  - foo.example.com
  http:
  - route:
    - destination:
        host: example.default.svc.domain.suffix
        port:
          number: 80
---
